lib/sensors/include/Sensor.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+class Sensor{
+  public:
+	//possible input pin general variables to include?
+	
+	//need to create a struct to pass into constructor?
+	Sensor() = default;
```

Comment:
Is the Sensor class supposed to be purely virtual? If so then remove the constructor and make all the virtual functions `= 0`